packages/twenty-front/src/modules/settings/data-model/validation-schemas/__tests__/settingsDataModelObjectAboutFormSchema.test.ts

```diff
@@ -0,0 +1,145 @@
+      },
+    },
+    {
+      title: 'validates input with optional fields omitted',
+      context: {
+        input: {
+          labelPlural: 'Labels Plural',
+          labelSingular: 'Label Singular',
+          namePlural: 'labelsPlural',
+          nameSingular: 'labelSingular',
+          isLabelSyncedWithName: false,
+        },
+        expectedSuccess: true,
+      },
+    },
+    {
+      title: 'validates input with trimmed labels',
+      context: {
+        input: {
+          ...validInput,
+          labelPlural: '   Labels Plural   ',
+          labelSingular: '  Label Singular  ',
+        },
+        expectedSuccess: true,
+      },
+    },
+  ];
+
+  const failsValidationTestsUseCase: EachTestingContext<{
+    input: Partial<Record<keyof SettingsDataModelObjectAboutFormValues, any>>;
+    expectedSuccess: false;
+  }>[] = [
+    {
+      title: 'fails when required fields are missing',
+      context: {
+        input: {
+          description: 'Only description',
+        },
+        expectedSuccess: false,
+      },
+    },
+    {
+      title: 'fails when names are not in camelCase',
+      context: {
+        input: {
+          ...validInput,
+          namePlural: 'Labels_Plural',
+          nameSingular: 'Label-Singular',
+        },
+        expectedSuccess: false,
+      },
+    },
+    {
+      title: 'fails when labels are empty strings',
+      context: {
+        input: {
+          ...validInput,
+          labelPlural: '',
+          labelSingular: '',
+        },
+        expectedSuccess: false,
+      },
+    },
+    {
+      title: 'fails when singular and plural labels are the same',
+      context: {
+        input: {
+          ...validInput,
+          labelPlural: 'Same Label',
+          labelSingular: 'Same Label',
+        },
+        expectedSuccess: false,
+      },
+    },
+    {
+      title: 'fails when singular and plural names are the same',
+      context: {
+        input: {
+          ...validInput,
+          namePlural: 'sameName',
+          nameSingular: 'sameName',
+        },
+        expectedSuccess: false,
+      },
+    },
+    {
+      title: 'fails when isLabelSyncedWithName is not a boolean',
+      context: {
+        input: {
+          ...validInput,
+          isLabelSyncedWithName: 'true',
+        },
+        expectedSuccess: false,
+      },
+    },
+    {
+      title: 'fails with invalid types for optional fields',
+      context: {
+        input: {
+          ...validInput,
+          description: 123,
+          icon: true,
+        },
+        expectedSuccess: false,
+      },
+    },
+  ];
+
+  test.each([...passingTestsUseCase, ...failsValidationTestsUseCase])(
+    '$title',
+    ({ context: { expectedSuccess, input } }) => {
+      const result = settingsDataModelObjectAboutFormSchema.safeParse(input);
+      expect(result.success).toBe(expectedSuccess);
+      if (!expectedSuccess) {
+        expect(result.error).toMatchSnapshot();
```

Comment:
I am new to snapshotting in JEST. Correct me if I am wrong, but this is JEST that created the folder __snapshot__ and its containg file.

Question: UI test tend to be flaky. even though I see in the doc the jest update function, it will break the CI. Are we ok and aware with implementing this ?

Reply:
> I am new to snapshotting in JEST. Correct me if I am wrong, but this is JEST that created the folder snapshot and its containg file.

Exactly

> Question: UI test tend to be flaky. even though I see in the doc the jest update function, it will break the CI. Are we ok and aware with implementing this ?

I think you're mistaking e2e tests and unit/integrations tests, in this tests suite we should never have flakiness

Reply:
True. But if we add a field to the form, do we want to break the test ? Answer could be yes, if we consider its a cruacial part. that's my point

Reply:
In my opinion yes, but before the test to fail, TypeScript would also raises regarding test definition itself saying: `dude your mocks does not relate my inner types`